Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -2953,6 +2954,30 @@ - (_WKRectEdge)_fixedContainerEdges
     return [self _sampledRightFixedPositionContentColor:_fixedContainerEdges];
 }
 
+- (void)_updateScrollGeometryWithContentOffset:(CGPoint)contentOffset contentSize:(CGSize)contentSize
+{
+    CGSize containerSize = [self frame].size;
+    auto contentInsets = _page->obscuredContentInsets();
+#if PLATFORM(IOS_FAMILY)
+    UIEdgeInsets cocoaInsets = UIEdgeInsetsMake(contentInsets.top(), contentInsets.left(), contentInsets.bottom(), contentInsets.right());
+#else
+    NSEdgeInsets cocoaInsets = NSEdgeInsetsMake(contentInsets.top(), contentInsets.left(), contentInsets.bottom(), contentInsets.right());
+#endif
+
+    RetainPtr oldScrollGeometry = _currentScrollGeometry;
+    RetainPtr newScrollGeometry = adoptNS([[WKScrollGeometry alloc] initWithContainerSize:containerSize contentInsets:cocoaInsets contentOffset:contentOffset contentSize:contentSize]);
+
+    if (oldScrollGeometry && [oldScrollGeometry isEqual:newScrollGeometry.get()])
+        return;
+
+    id<WKUIDelegateInternal> uiDelegate = (id<WKUIDelegateInternal>)self.UIDelegate;
+    if (![uiDelegate respondsToSelector:@selector(_webView:geometryDidChange:)])
+        return;
+
+    [uiDelegate _webView:self geometryDidChange:newScrollGeometry.get()];
+    _currentScrollGeometry = newScrollGeometry;
```

Comment:
Can we swap the order of these lines? It's more idiomatic to update state prior to making delegate calls.